php_phongo.c

```diff
@@ -850,7 +852,12 @@ int phongo_execute_command(mongoc_client_t* client, php_phongo_command_type_t ty
 			return false;
 	}
 	if (!result) {
-		phongo_throw_exception_from_bson_error_t(&error TSRMLS_CC);
+		if (error.domain == MONGOC_ERROR_SERVER || error.domain == MONGOC_ERROR_WRITE_CONCERN) {
+			phongo_add_exception_prop(ZEND_STRL("resultDocument"), (zval*) &reply TSRMLS_CC);
```

Comment:
If you look at the definition for `phongo_add_exception_prop()` in `phongo_compat.c`, you'll notice that it is a no-op unless `EG(exception)` is non-zero (i.e. an exception has been thrown). This function should be called after we throw the exception on the next line. That explains why the `resultDocument` property was null in the tests.

Independent of that, we can't simply cast a `bson_t *` to a `zval *` (although it'd be great if BSON conversation was that easy). We'll need to use `php_phongo_bson_to_zval()` or `php_phongo_bson_to_zval_ex()` to convert the BSON document to a PHP zval. Since we're not supporting custom type maps here, and this isn't debug output where we'd convert everything to arrays, I'll suggest we use `php_phongo_bson_to_zval()` and its default type map. For prior art, see: https://github.com/mongodb/mongo-php-driver/blob/0a745ee8dee43ca7952afea6217678302e13b5bc/src/MongoDB/Command.c#L187-L192

To make life simpler, you should be able to reuse the `return_value` variable instead of conditionally declaring a zval for PHP 5.x and 7+. This is similar to what we do in `phongo_execute_bulk_write()`, where `return_value` (always a `zval *`, regardless of PHP version) is initialized as a WriteResult object and later assigned to an exception property.

Therefore, after using `return_value` as the destination for BSON conversion, it should then we suitable to use with `phongo_add_exception_prop()` directly.